authentication.md

```diff
@@ -139,3 +139,31 @@ Status codes:
 * 400 Bad Request - if the X-On-Behalf-Of header doesn't contain a valid EPerson UUID
 * 403 Forbidden - if you are not authorized to act on behalf of the given user
 * Any status code of the functionality being used
+
+
+## Request download token
+
+** POST /api/authn/download-token **
+
+When clicking on a link to download a protected file in the UI no authentication header will be sent along. This endpoint can provide a short lived token (MAX 2 seconds) that the UI can append to file downloads.
+ 
+The token follows the "JSON Web Token structure", same as the login tokens.
+  
+ ```
+ curl -v -X POST https://{dspace-server.url}/api/authn/download-token -H "Authorization: Bearer eyJhbG...COdbo"
+ ```
+ 
+ ```json
+{
+  "token": "eyJhbGciOiJIUzI1NiJ9.eyJlaWQiOiJjZDgyNGE2MS05NWJlLTRlMTYtYmNjZC01MWZlYTI2NzA3ZDAiLCJzZyI6W10sImV4cCI6MTU5MDQxMzUwNn0.XRK4ldh9l4My45gJzLtcW97hVUpbtM5oAQsxuQ2V37c",
+  "_links": {
+    "self": {
+      "href": "http://${dspace-server.url}/api/authn/download-token"
+    }             
+  }
```

Comment:
no self link is needed here, this object is not addressable

Reply:
This is something that @artlowel has always urged me to include in our rest responses, as it is a requirement from the HAL spec. This way angular can parse all of the rest responses in a similar manner.